### PR TITLE
Properly handle multi-line blockquotes

### PR DESCRIPTION
**Pull request by golmansax**
Multi-line blockquotes weren't being parsed correctly so I took a stab at solving it.  Not sure if I found the correct solution though.

Issue:
```
> Quote1
>
> Quote2
```
Was incorrectly creating a blockquote with text `"Quote1Quote2"`

### CODE REVIEW

**Comment by tommoor** [comment]
Hey @golmansax – thanks for taking a stab, if you're able to get the existing and new tests passing I'll be happy to merge a solution.

**Comment by golmansax** [comment]
@tommoor Was able to get this to pass CI, let me know what you think!

**Comment by tommoor** [comment]
The changes make sense, although I do think ideally the text inside the multiline blockquote should be split into individual paragraphs nodes.

I'm trying to get this to work correctly with https://github.com/outline/rich-markdown-editor but haven't had success thus far.

**Comment by golmansax** [comment]
Hm, I think you're right.  Let me try again.

**Comment by golmansax** [comment]
@tommoor Was able to get it working so that paragraphs are nested within blockquotes.  I also tested it on a live project so this time I feel better about the result.  However, there's a check for `obj.type === 'block-quote'` that I'm not too proud of.

What do you think?

File: src/renderer.js [hunk 1]
```diff
@@ -207,10 +208,13 @@ class Markdown {
       });
     }
 
-    let children = node.nodes.map(node => this.serializeNode(node, document));
-    children = children.flatten().length === 0
-      ? ""
-      : children.flatten().join("");
```
Comment (tommoor) [inline src/renderer.js]: Is there a particular reason you removed the flatten usage?

**Comment by tommoor** [comment]
Thanks! This works pretty well in my testing, one thing that's a bit strange is that having a list in a blockquote appears to add extra paragraphs above on serialize/deserialize – I wonder if that's related to the special case you had to add?

File: src/renderer.js (hunk 1 above)
Comment (golmansax) [reply src/renderer.js]: Yeah.  The flatten logic was essentially joining everything with `""`.  To get the multi-line blockquote use case to work, we needed to use `""` to join the inner elements but use `"\n"` to join nodes.

**Comment by golmansax** [comment]
Hmm let me add a test for that and debug.

I'm also working on https://github.com/tommoor/slate-md-serializer/pull/11 in conjunction because my project requires all of the empty paragraphs to be removed, so my live testing probably won't catch empty paragraph issues.

**Comment by golmansax** [comment]
@tommoor I added a test in 3c9c5e8 which I think describes the list in a blockquote example you mentioned.  I didn't see the issue with empty paragraphs being added.  See anything different between the test and the use case you were describing?

**Comment by tommoor** [comment]
It may have been multi-line quote specific - try creating a multiline quote with a list within.

🙏

**Comment by golmansax** [comment]
Added a second list-item to the quote in 64cc629 but it still has no empty paragraphs.

**Comment by tommoor** [comment]
Tested in all the scenarios I can think of, great bug fix – sorry for the delay in merging 😄
